### PR TITLE
chore: first implementation for Textarea component

### LINKED ISSUE #136: Textarea component
Notes Section

The uppercase "Notes" should be a configurable title. The lorem-ipsum text is the user input.

### PR DESCRIPTION
**Pull request by florian-sabonchi**
Closes #136

### CODE REVIEW

**Comment by florian-sabonchi** [comment]
image

image

File: tasks/components/user_input/Textarea.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { tw } from '@helpwave/common/twind'
+
+type TextareaProps = {
+  headline : string,
+  placeholder : string,
+  message : string,
+  id : string
```
Comment (use-to) [inline tasks/components/user_input/Textarea.tsx:7]: please fix this formatting

File: tasks/components/user_input/Textarea.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+}
+export const Textarea = ({ placeholder, headline, message, id } : TextareaProps) => {
```
Comment (use-to) [inline tasks/components/user_input/Textarea.tsx:9]: use the order as above

File: tasks/components/user_input/Textarea.tsx (hunk 1 above)
Comment (use-to) [inline tasks/components/user_input/Textarea.tsx:7]: All of these parameters should be optional

Comment (use-to) [reply tasks/components/user_input/Textarea.tsx:7]: Would be nice to have an min and max length parameter

File: tasks/components/user_input/Textarea.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  return (
+    <div className={tw('relative shadow border-1 border-gray-300 rounded-lg')}>
+      <label className={tw('mx-3 mt-3  block text-gray-700 font-bold')} htmlFor={id}>
+        {headline}
+      </label>
+      <textarea defaultValue={message} className={tw('pt-0 border-transparent focus:border-transparent focus:ring-0 h-32 resize-none  appearance-none border w-full text-gray-700 leading-tight focus:outline-none')} id={id} placeholder={placeholder}>
```
Comment (use-to) [inline tasks/components/user_input/Textarea.tsx:15]: use our primary color as the focus border color

Comment (use-to) [reply tasks/components/user_input/Textarea.tsx:15]: hide the potential scrollbar or change the optical appearance

File: tasks/components/user_input/Textarea.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
```
Comment (use-to) [inline tasks/components/user_input/Textarea.tsx:16]:
```suggestion
```

**Review by use-to — CHANGES_REQUESTED** [review]

File: tasks/components/user_input/Textarea.tsx (hunk 1 above)
Comment (DasProffi) [reply tasks/components/user_input/Textarea.tsx:7]: pass `{...props}` to the input element, see [this](https://github.com/helpwave/web/blob/main/tasks/components/user_input/Input.tsx) for a reference

Comment (DasProffi) [inline tasks/components/user_input/Textarea.tsx:7]:
```suggestion
  headline?: string,
  placeholder?: string,
  message?: string,
  id?: string,
  onChange: (text: string) => void,
  props?: TextareaHTMLAttributes<HTMLTextAreaElement>,
```

**Review by DasProffi — CHANGES_REQUESTED** [review]

File: tasks/components/user_input/Textarea.tsx (hunk 1 above)
Comment (use-to) [reply tasks/components/user_input/Textarea.tsx:7]: thats a good idea

File: tasks/components/user_input/Textarea.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  /* style of the scrollbar of the textarea */
+  textarea::-webkit-scrollbar {
+    @apply w-1;
+  }
+
+  textarea::-webkit-scrollbar-thumb {
+    @apply rounded-lg bg-hw-primary-600;
+  }
```
Comment (DasProffi) [inline tasks/components/user_input/Textarea.tsx:26]: The Scrollbar styling should be applied globally and not just for this component's scrollbar.

You can just omit it here, and we will open an additional issue for that

File: tasks/components/user_input/Textarea.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+`
+export const Textarea = ({ headline, placeholder, message, id } : TextareaProps) => {
+  return (
+    <div className={tw(globalStyles)}>
+      <div className={`textarea-wrapper ${tw('relative shadow border-1 border-gray-300 rounded-lg')}`}>
+        <label className={tw('mx-3 mt-3 block text-gray-700 font-bold')} htmlFor={id}>
+          {headline}
+        </label>
+        <textarea defaultValue={message} className={tw('pt-0 border-transparent focus:border-transparent focus:ring-0 h-32 resize-none appearance-none border w-full text-gray-700 leading-tight focus:outline-none')} id={id} placeholder={placeholder}>
```
Comment (DasProffi) [inline tasks/components/user_input/Textarea.tsx:35]:
```suggestion
        <textarea 
          defaultValue={message}
          className={tw('pt-0 border-transparent focus:border-transparent focus:ring-0 h-32 resize-none appearance-none border w-full text-gray-700 leading-tight focus:outline-none')}
          id={id}
          placeholder={placeholder}
        >
```

**Review by DasProffi — CHANGES_REQUESTED** [review]

File: tasks/components/user_input/Textarea.tsx (hunk 5 above)
Comment (florian-sabonchi) [reply tasks/components/user_input/Textarea.tsx:26]: Sound good to me

**Review by DasProffi — APPROVED** [review]

File: tasks/components/user_input/Textarea.tsx (hunk 5 above)
Comment (use-to) [reply tasks/components/user_input/Textarea.tsx]: ++ #163

File: tasks/components/user_input/Textarea.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
+import { css, tw } from '@helpwave/common/twind'
+import type { TextareaHTMLAttributes } from 'react'
+
+type TextareaProps = {
+  headline?: string,
+  placeholder?: string,
+  message?: string,
+  id?: string,
+  onChange: (text: string) => void,
```
[the remark this answers is not in the document]
Comment (use-to) [reply tasks/components/user_input/Textarea.tsx:9]: default noop

File: tasks/components/user_input/Textarea.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  }
+`
+export const Textarea = ({ headline, placeholder, message, id } : TextareaProps) => {
+  return (
+    <div className={tw(globalStyles)}>
+      <div className={`textarea-wrapper ${tw('relative shadow border-2 border-gray-300 rounded-lg')}`}>
+        <label className={tw('mx-3 mt-3 block text-gray-700 font-bold')} htmlFor={id}>
+          {headline}
+        </label>
+        <textarea
```
Comment (use-to) [inline tasks/components/user_input/Textarea.tsx:35]: where do you handle `props`?

Comment (use-to) [inline tasks/components/user_input/Textarea.tsx:35]: where do you handle the `onChange`?

File: tasks/components/user_input/Textarea.tsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
+import { css, tw } from '@helpwave/common/twind'
+import type { TextareaHTMLAttributes } from 'react'
+
+type TextareaProps = {
+  headline?: string,
+  placeholder?: string,
+  message?: string,
+  id?: string,
+  onChange: (text: string) => void,
+  props?: TextareaHTMLAttributes<HTMLTextAreaElement>
```
Comment (use-to) [inline tasks/components/user_input/Textarea.tsx:10]: Why would you put it here?

**Review by use-to — CHANGES_REQUESTED** [review]
From a design perspective I think you nailed it, but there is a big lack of functionality.
You should try to test it by yourself... I don't want to be the first one (as a reviewer) putting this component inside of an actual view.
Why was this pr already approved here?

**Comment by use-to** [comment]
Maybe take a look at other components like `Input.tsx`! @florian-sabonchi

**Comment by MaxSchaefer** [comment]
To streamline the follow-up changes:
- [ ] `props.onChange` should be optional with a default value of `noop`
- [ ] changes of the `textarea` should be propagated to the `onChange` callback
- [ ] The headline is optional, this should be handled inside the component as well. `!headline -> dont render the label`
- [ ] Take [ControlledInput](https://github.com/helpwave/web/blob/main/tasks/components/user_input/Input.tsx#L27) as an example. You can use `value` instead of `defaultValue` on the `textarea`. Most of the time, we will passing some `useState` reference.

As @use-to already said, the styling is very nice! Keep going.

**Comment by use-to** [comment]
> 

What's way better explained then I did.

**Review by use-to — APPROVED** [review]

**Comment by MaxSchaefer** [comment]
@use-to @florian-sabonchi https://github.com/helpwave/web/issues/166

**Pull request merged**